pycaret/internal/pipeline.py

```diff
@@ -79,6 +80,42 @@ def __init__(self, steps, *, memory=None, verbose=False):
         self._fit_vars = set()
         self._feature_names_in = None
 
+    def __getstate__(self):
+        try:
+            state = super().__getstate__()
+        except AttributeError:
+            state = self.__dict__.copy()
+
+        from importlib import import_module
+
+        versions = {"pycaret": __version__}
+        ml_modules = [
+            ("sklearn", "sklearn"),
+            ("imblearn", "imblearn"),
+            ("pyod", "pyod.version"),
+            ("category-encoders", "category_encoders"),
+            ("lightgbm", "lightgbm"),
+        ]
+        for name, import_name in ml_modules:
+            try:
+                module = import_module(import_name)
+                versions[name] = module.__version__
+            except (AttributeError, ImportError, OSError):
+                pass
+
+        return dict(state.items(), _pycaret_version=versions)
+
+    def __setstate__(self, state):
+        pickle_versions = state.get("_pycaret_version", {})
+        if pickle_versions["pycaret"] != __version__:
```

Comment:
This will fail if pickle_versions is `{}` right?

```suggestion
        if pickle_versions.get("pycaret") != __version__:
```